adagrams/game.py

```diff
@@ -1,11 +1,181 @@
+        'J': 8, 
+        'K': 5, 
+        'L': 1, 
+        'M': 3, 
+        'N': 1, 
+        'O': 1, 
+        'P': 3, 
+        'Q': 10, 
+        'R': 1, 
+        'S': 1, 
+        'T': 1, 
+        'U': 1, 
+        'V': 4, 
+        'W': 4, 
+        'X': 8, 
+        'Y': 4, 
+        'Z': 10
+}
+
 def draw_letters():
-    pass
+
+    letter_bank = []
+    temp_letter_pool = copy.deepcopy(LETTER_POOL)
+
+    while len(letter_bank) < 10:        
+        letter = random.choice(list(temp_letter_pool.keys()))
+        if temp_letter_pool[letter] > 0:
+            temp_letter_pool[letter] -= 1
+            letter_bank.append(letter)
+        else:
+            continue
+
+    return letter_bank
 
 def uses_available_letters(word, letter_bank):
-    pass
+    temp_letter_bank = copy.deepcopy(letter_bank)
+
+    for letter in word:
+        if letter.upper() in temp_letter_bank:
+            temp_letter_bank.remove(letter.upper())
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    score = 0
+    
+    for letter in word:
+        score += SCORE_DICT.get(letter.upper())
+    
+    if len(word) >= 7:
+        score += 8
+    
+    return score
+
+def replace_highest_values(high_word, high_score, new_word, new_score):
+    high_word = new_word
+    high_score= new_score
+    
+    return high_word, high_score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    highest_score = 0
+    highest_word = ""
+    for word in word_list:
+        score_of_word = score_word(word)
+        if score_of_word > highest_score:
+            highest_word, highest_score = replace_highest_values(highest_word, highest_score, word, score_of_word)
+        elif score_of_word == highest_score:
+            if len(word) == 10 and len(highest_word) !=10:
+                highest_word, highest_score = replace_highest_values(highest_word, highest_score, word, score_of_word)
+            elif len(word) <len(highest_word) and len(highest_word) != 10:
+                highest_word, highest_score = replace_highest_values(highest_word, highest_score, word, score_of_word)
+        print(score_of_word)
+    return tuple([highest_word, highest_score])
+
+
+
+
+
+# def get_highest_word_score_working(word_list):
+#     highest_score = 0
+#     highest_word = ""
+#     for word in word_list:
+#         score_of_word = score_word(word)
+#         if score_of_word > highest_score:
+#             highest_score = score_of_word
+#             highest_word = word
+#         elif score_of_word == highest_score:
+#             if len(word) == 10 and len(highest_word) !=10:
+#                 highest_score = score_of_word
+#                 highest_word = word
+#             elif len(word) <len(highest_word) and len(highest_word) != 10:
+#                 highest_score = score_of_word
+#                 highest_word = word
+#         print(score_of_word)
+#     return tuple([highest_word, highest_score])
```

Comment:
Thanks for letting me see what y'all originally had. I actually love seeing where y'all's refactoring happened. In industry, feel free to store these previous solutions elsewhere if you'd like, or just make sure to delete them from the final code!